src/plan.rs

```diff
@@ -0,0 +1,504 @@
+    Pk: MiniscriptKey + ToPublicKey,
+{
+    fn lookup_ecdsa_sig(&self, pk: &Pk) -> bool {
+        Satisfier::lookup_ecdsa_sig(self, pk).is_some()
+    }
+
+    fn lookup_tap_key_spend_sig(&self, _: &Pk) -> bool {
+        Satisfier::lookup_tap_key_spend_sig(self).is_some()
+    }
+
+    fn lookup_tap_leaf_script_sig(&self, pk: &Pk, leaf_hash: &TapLeafHash) -> bool {
+        Satisfier::lookup_tap_leaf_script_sig(self, pk, leaf_hash).is_some()
+    }
+
+    fn lookup_raw_pkh_pk<Ctx: ScriptContext>(&self, hash: &hash160::Hash) -> Option<usize> {
+        Satisfier::lookup_raw_pkh_pk(self, hash).map(|p| Ctx::pk_len(&p))
+    }
+
+    fn lookup_raw_pkh_x_only_pk(&self, hash: &hash160::Hash) -> bool {
+        Satisfier::lookup_raw_pkh_x_only_pk(self, hash).is_some()
+    }
+
+    fn lookup_raw_pkh_ecdsa_sig<Ctx: ScriptContext>(&self, hash: &hash160::Hash) -> Option<usize> {
+        Satisfier::lookup_raw_pkh_ecdsa_sig(self, hash).map(|(p, _)| Ctx::pk_len(&p))
+    }
+
+    fn lookup_raw_pkh_tap_leaf_script_sig(&self, hash: &(hash160::Hash, TapLeafHash)) -> bool {
+        Satisfier::lookup_raw_pkh_tap_leaf_script_sig(self, hash).is_some()
+    }
+
+    fn lookup_sha256(&self, hash: &Pk::Sha256) -> bool {
+        Satisfier::lookup_sha256(self, hash).is_some()
+    }
+
+    fn lookup_hash256(&self, hash: &Pk::Hash256) -> bool {
+        Satisfier::lookup_hash256(self, hash).is_some()
+    }
+
+    fn lookup_ripemd160(&self, hash: &Pk::Ripemd160) -> bool {
+        Satisfier::lookup_ripemd160(self, hash).is_some()
+    }
+
+    fn lookup_hash160(&self, hash: &Pk::Hash160) -> bool {
+        Satisfier::lookup_hash160(self, hash).is_some()
+    }
+
+    fn check_older(&self, s: Sequence) -> bool {
+        Satisfier::check_older(self, s)
+    }
+
+    fn check_after(&self, l: LockTime) -> bool {
+        Satisfier::check_after(self, l)
+    }
+}
+
+/// Representation of a particular spending path on a descriptor. Contains the witness template
+/// and the timelocks needed for satisfying the plan.
+/// Calling `get_plan` on a Descriptor will return this structure,
+/// containing the cheapest spending path possible (considering the `Assets` given)
+#[derive(Debug, Clone)]
+pub struct Plan {
+    /// This plan's witness template
+    pub template: WitnessTemplate<Placeholder<DefiniteDescriptorKey>>,
+    /// The absolute timelock this plan uses
+    pub absolute_timelock: Option<LockTime>,
+    /// The relative timelock this plan uses
+    pub relative_timelock: Option<Sequence>,
+
+    pub(crate) desc_type: DescriptorType,
+}
+
+impl Plan {
+    /// Returns the witness version
+    pub fn witness_version(&self) -> Option<WitnessVersion> {
+        self.desc_type.segwit_version()
+    }
+
+    /// The weight, in witness units, needed for satisfying this plan (includes both
+    /// the script sig weight and the witness weight)
+    pub fn satisfaction_weight(&self) -> usize {
+        self.witness_size() + self.scriptsig_size() * 4
```

Comment:
Witness size is what I refer as the available assets weight. But what is scriptsig size and why we need addition of both.